test/apps/iodemo/io_demo.cc

```diff
@@ -782,9 +851,12 @@ class DemoClient : public P2pDemoCommon {
             ++(*_io_counter);
             ++(*_server_io_counter);
             if (_validate && (status == UCS_OK)) {
-                iomsg_t *msg = reinterpret_cast<iomsg_t*>(_buffer);
-                validate(msg, _sn, _buffer_size);
                 validate(*_iov, _sn);
+
+                if (_meta_comp_counter != 0) {
```

Comment:
I would expect to do validation when whole the op is completed?

Reply:
so, we should just change the pointer for validation but don't skip it at all?

Reply:
added validation